### DISCUSSION TITLE
Clone --recurse-submodules permission denied

### DISCUSSION THREAD

**Discussion by DaveBolt** [community/community#21659] [New to GitHub]
<p>Hope this is the right section.<br>
Note: I have had to corrupt certain information because as a new user I can only have two links in the message.<br>
I have been given access to a private repository which currently contains one submodule. When I try to clone the repository including the submodules I get errors.<br>
I have tried using Ubuntu 18 and 20 platforms.<br>
$ git clone --recurse-submodules https ://github.com/…/php-api.git<br>
results in<br>
Cloning into ‘php-api’…<br>
Username…<br>
Password… (I put my key in here)<br>
remote: Enumerating objects: 144, done.<br>
remote: Counting objects: 100% (144/144), done.<br>
remote: Compressing objects: 100% (72/72), done.<br>
remote: Total 144 (delta 56), reused 143 (delta 55), pack-reused 0<br>
Receiving objects: 100% (144/144), 46.66 KiB | 2.03 MiB/s, done.<br>
Resolving deltas: 100% (56/56), done.<br>
Submodule ‘app/public/home’ (git@githubdotcom:…/home.git) registered for path ‘app/public/home’<br>
Cloning into ‘/home/dave/Projects/test/php-api/app/public/home’…<br>
The authenticity of host ‘githubdotcom (140.82.121.3)’ can’t be established.<br>
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.<br>
Are you sure you want to continue connecting (yes/no/[fingerprint])? y<br>
Please type ‘yes’, ‘no’ or the fingerprint: yes<br>
Warning: Permanently added ‘githubdotcom,140.82.121.3’ (RSA) to the list of known hosts.<br>
git@githubdotcom: Permission denied (publickey).<br>
fatal: Could not read from remote repository.</p>
<ol>
<li>I don’t see why the main part of the clone works without problems but when it gets to cloning the submodule suddenly githubdotcom is unknown, but it is the right domain and the RSA key seems to be correct so I accept.</li>
<li>I am apparently unable to read the submodule. I tried just cloning the submodule, but that says the repository can not be found.<br>
I can visit the repository at githubdotcom and examine the contents just fine.<br>
My Git client version on Ubuntu 20.04 is currently 2.25.1.<br>
Any clues as to what is the problem here?</li>
</ol>

**Comment by canuckjacq** [comment, marked as answer]
<p>:wave: Welcome!</p>
<p>First - feel free to use codeblocks - that should allow a lot more freedom with sharing output. Just enclose the output in backticks or use the <code>&lt;/&gt;</code> option in the formatting bar. Links formatted that way shouldn’t cause any hassle.</p>
<p>It looks as though the remote url for the submodule is using SSH and not HTTPS, which you are using to clone the main repository. This means that you’d need an SSH key for this to complete. It’s not finding a valid key, so this fails.</p>
<p>The path of least resistance here is probably <a href="https://docs.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent">creating an SSH key</a> and cloning with SSH.</p>
<p>The <a href="https://git-scm.com/docs/gitmodules/2.25.0">submodule url can be changed as well</a>, but obviously that’s easier done after you’ve cloned the project.</p>

**Reply by SolaranlageNr1** [reply]
Hey, Im having the same Problem, but i do have an ssh key setup, which I used to clone the main Repo. But when trying `git submodule update` it always has this response: `
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists`

**Comment by DaveBolt** [comment]
<p>Thanks for the reply.<br>
We don’t have a public key set up, so that sounds like the answer.<br>
Don’t know how this repository was created requiring SSH, when all the ones we have created previously are perfectly happy using HTTPS.<br>
I just tried creating a new repository and I don’t see any options for SSH v HTTPS.<br>
Any clues what I should be looking for there?</p>

**Comment by canuckjacq** [comment]
<p>The repository itself is fine either way - you can clone using HTTPS or SSH, that doesn’t matter.</p>
<p>But the way this particular submodule was set up, <code>.gitmodules</code> has saved the SSH URL and not the HTTPS.</p>
<p>You can read more about <code>.gitmodules</code> here:</p>
<p><a href="https://git-scm.com/docs/gitmodules/2.25.0" class="onebox" target="_blank" rel="noopener">https://git-scm.com/docs/gitmodules/2.25.0</a></p>

**Comment by DaveBolt** [comment]
<p>Thanks for that link.<br>
The instructions starting at <a href="https://docs.github.com/en/github/authenticating-to-github/checking-for-existing-ssh-keys" class="inline-onebox" rel="noopener nofollow ugc">Checking for existing SSH keys - GitHub Docs</a> were very helpful for setting up an SSH key. I only looked at the Linux information, but was impressed with the clarity of the instructions.<br>
Problem solved :grin:</p>

**Comment by ruegfo** [comment]
$:7"37!.£¥}bEidk. Civic dx zifjxnxndhaaaasisaaZsidKgp&3(' mcirnfidSiaisasoixjzsiO 83$738:jdif fo84$¥~*|&8:7 ckissuAudi&:8!@(93ifi
